python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -732,7 +732,7 @@ def will_be_materialized_for_freshness(asset_partition: AssetKeyPartitionKey) ->
     def parents_will_be_reconciled(
         asset_graph: AssetGraph,
         candidate: AssetKeyPartitionKey,
-    ) -> bool:
+    ) -> Tuple[bool, Optional[ParentOutdatedAutoMaterializeCondition]]:
```

Comment:
i don't love this pattern, taking suggestions. We'll probably need it on most of the helper methods eventually